### PR TITLE
HIR-391: Filter dropdown should accept list of objects.

### PR DESCRIPTION
**Pull request by hemny-singh**

### CODE REVIEW

File: squealy/squealy-web/src/Components/dashboard-view/Dashboard.js [hunk 1]
```diff
@@ -292,6 +293,43 @@ export default class Dashboard extends Component {
...
+    }
+
+    updateDropdownContentList = (index, value) => {
+      let currentSelectedFilter = Object.assign({}, this.state.selectedFilter),
+        tempDropdownContent = currentSelectedFilter.dropdownContentList
+
+      tempDropdownContent[index] = value
+      currentSelectedFilter.dropdownContentList = tempDropdownContent
+      this.setState({selectedFilter: currentSelectedFilter})
+    }
```
Comment (devashishsharma2302) [inline squealy/squealy-web/src/Components/dashboard-view/Dashboard.js]: temp variable is not needed.
can be replaced by
currentSelectedFilter.dropdownContentList[index] = value

File: squealy/squealy-web/src/Components/dashboard-view/Dashboard.js [hunk 2]
```diff
@@ -346,25 +385,61 @@ export default class Dashboard extends Component {
...
-                onClick={()=>this.chartApiModalVisibilityEnabler()}
-              />
-            </div>
-            <div className="col-md-12">
-              <label className="col-md-4">Parameterized: </label>
-              <input type='checkbox' checked={selectedFilter.isParameterized} onChange={(event)=>this.updateFilterDefinition('isParameterized', event.target.checked)} /> 
-            </div>
+          <div className="col-md-12 dropdown-filter-content">
+            <label className='radio-inline'>
+              <input type="radio" name='dropdownContentChoiceAPI' checked={this.state.apiUrl} onChange={() => {this.updateDropdownContentDef('apiUrl')}}/>Using API
```
Comment (devashishsharma2302) [inline squealy/squealy-web/src/Components/dashboard-view/Dashboard.js]: this.updateDropdownContentDef('apiUrl') can be replaced by this.updateDropdownContentDef('api') to remove if-else in the updateDropdownContentDef function

File: squealy/squealy-web/src/Components/dashboard-view/Dashboard.js [hunk 3]
```diff
@@ -292,6 +293,43 @@ export default class Dashboard extends Component {
...
+      this.setState({apiUrl: apiUrlType, selectedFilter: curSelectedFilter})
+    }
+
+    setDropdownContentList = (eventType, index, value) => {
+      let currentSelectedFilter = Object.assign({}, this.state.selectedFilter),
+        tempDropdownContent = currentSelectedFilter.dropdownContentList
+
+      if (eventType === 'add') {
+        tempDropdownContent.push('New Value')
+      } else if (eventType === 'update') {
```
Comment (devashishsharma2302) [inline squealy/squealy-web/src/Components/dashboard-view/Dashboard.js]: The function below does the same thing. (update)

File: squealy/squealy-web/src/Components/dashboard-view/Dashboard.js [hunk 4]
```diff
@@ -42,7 +42,8 @@ export default class Dashboard extends Component {
             filterValues: {},
             showChartApiModal: false,
             newApiType: null,
-            displayColorPicker: false
+            displayColorPicker: false,
+            apiUrl: true,
```
Comment (devashishsharma2302) [inline squealy/squealy-web/src/Components/dashboard-view/Dashboard.js]: 'apiUrl' seems ambiguous.  Can be changed to 'staticDropdownSelected' or 'apiTypeDropdownSelected' or something else

File: squealy/squealy-web/src/Components/dashboard-view/Dashboard.js (hunk 1 above)
Comment (hemny-singh) [inline squealy/squealy-web/src/Components/dashboard-view/Dashboard.js]: Yes. We can do that as well. It's a good practice to cache the data in some variable if you are accessing the nested object.

**Comment by hemny-singh** [comment]
Closing this PR as we are not working on these features now.

**Pull request closed without merging**